libraries/cmake/source/augeas/CMakeLists.txt

```diff
@@ -0,0 +1,171 @@
+# Copyright (c) 2014-present, Facebook, Inc.
+# All rights reserved.
+#
+# This source code is licensed in accordance with the terms specified in
+# the LICENSE file found in the root directory of this source tree.
+
+function(augeasMain)
```

Comment:
We now depend on Bison and Flex to generate the parser for Augeas lenses; if this is not acceptable, we could pre-generate the files and commit them

Reply:
What are the dependencies of that generation?
Meaning, why it's generated on the fly?
If there isn't something system related, I would just commit the pre-generated files.

Reply:
The only dependencies are the definition files describing the Augeas language; the .c/.cpp files are usually always generated inside the binary directory, since they are considered ephemeral (i.e. just like a build artifact).

Some examples (CMake also provides helper functions for some of these) include Thrift (but not how it's used within osquery), Protobuf and gRPC. Bison and Flex do have, in fact, CMake helpers that output to the binary folder (but I didn't use them here due to a bug).

I'd say the correct approach is undoubtedly to always generate the code at build time in the binary directory and only version the definition files, but in the case of osquery we may choose 
 to eliminate as many external dependencies as possible and bake everything

Reply:
I've moved the dependencies in the docker image, I'll push as soon as the test with binutils is done.